pkg/provisioner/ironic/ironic.go

```diff
@@ -1167,12 +1166,26 @@ func (p *ironicProvisioner) Deprovision() (result provisioner.Result, err error)
 		p.log.Info("cleaning")
 		result.Dirty = true
 		result.RequeueAfter = deprovisionRequeueDelay
+		stepClean, stepCleanOk := ironicNode.CleanStep["step"]
+
+		if stepCleanOk == true {
+			p.host.Status.Provisioning.Step = fmt.Sprintf("%v", stepClean)
```

Comment:
According to the [API docs](https://docs.openstack.org/api-ref/baremetal/?expanded=show-node-details-detail) the clean step is a string. Why do we need the Sprintf() call? Is the go value not a string?

Reply:
CleanStep is of type 'map[string]interface{}' and thus stepClean is of type 'interface{}'.
p.host.Status.Provisioning. Step is 'string' and I found Sprintf() to be a convenient way for type cast.

Reply:
Are those the types gophercloud uses? I wonder if we should change the library upstream if we know the value will only ever be a string.

Reply:
Yes the types are from gophercloud.
I don't have an answer for that if 'step' will be a string in the future. My proposal is to merge this PR and possibly have a separate task for the typing.